src/Microsoft.Extensions.Logging.Console/ConsoleLogger.cs

```diff
@@ -97,99 +97,103 @@ public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Except
         {
-            // check if the message has any new line characters in it and provide the padding if necessary
-            message = ReplaceMessageNewLinesWithPadding(message);
-            var logLevelColors = GetLogLevelConsoleColors(logLevel);
-            var loglevelString = GetLogLevelString(logLevel);
+            var logLevelColors = default(ConsoleColors);
+            var logLevelString = string.Empty;
+            var logIdentifier = string.Empty;
+            var scopeInformation = string.Empty;
+            var exceptionText = string.Empty;
+            var printLog = false;
 
             // Example:
             // INFO: ConsoleApp.Program[10]
             //       Request received
-
-            lock (_lock)
+            if (!string.IsNullOrEmpty(message))
             {
-                // log level string
-                WriteWithColor(
-                    logLevelColors.Foreground,
-                    logLevelColors.Background,
-                    loglevelString,
-                    newLine: false);
-
+                logLevelColors = GetLogLevelConsoleColors(logLevel);
+                logLevelString = GetLogLevelString(logLevel);
                 // category and event id
-                // use default colors
-                WriteWithColor(
-                    DefaultConsoleColor,
-                    DefaultConsoleColor,
-                    _loglevelPadding + logName + $"[{eventId}]",
-                    newLine: true);
-
+                logIdentifier = _loglevelPadding + logName + "[" + eventId + "]";
                 // scope information
                 if (IncludeScopes)
                 {
-                    var scopeInformation = GetScopeInformation();
+                    scopeInformation = GetScopeInformation();
+                }
+                // message
+                message = _messagePadding + ReplaceMessageNewLinesWithPadding(message);
+                printLog = true;
+            }
+
+            // Example:
+            // System.InvalidOperationException
+            //    at Namespace.Class.Function() in File:line X
+            if (exception != null)
+            {
+                // exception message
+                exceptionText = exception.ToString();
+                printLog = true;
+            }
+
+            if (printLog)
+            {
+                lock (_lock)
+                {
+                    if (!string.IsNullOrEmpty(logLevelString))
+                    {
+                        // log level string
+                        Console.Write(
+                            logLevelString,
+                            logLevelColors.Background,
+                            logLevelColors.Foreground);
+                    }
+
+                    // use default colors from here on
+                    if (!string.IsNullOrEmpty(logIdentifier))
+                    {
+                        Console.WriteLine(
```

Comment:
You **_could_** put this in the previous if check since if `logLevelString` is set then `logIdentifier` is set too, but that might be a bit too much :)